### ISSUE TITLE
Allow users to unstack all the files at once

### ISSUE THREAD

**Issue by alexislefebvre** [photoprism/photoprism#1178]
Today, if we scan files while a stacking parameter is enabled, we have stacked files and we can't unstack them easily, it requires to restart the indexing: https://github.com/photoprism/photoprism/issues/1167#issuecomment-812636775

It would be nice to allow users to unstack all files at once (and then to be able to enable the stacking parameters and index them again to stack them, but that's another issue).

Alternatively, launching a scan with or without the stacking parameters could create or delete the current stacks, so that the parameters don't work only during the initial scan.

Related but different issues:
-  Allow mass unstacking. #964
-  UX: Add "Unstack All" Button #667

**Comment by lastzero** [comment]
Note that the outcome of unstacking may depend on the order in which you do it. In order to unstack files, they may need to be renamed as well if they share the same file name (prefix), like photo.jpg and photo.raw. In case of non-visual sidecar files like photo.xmp, unstacking is not possible at all as it doesn't make any sense.

TL;DR: Unstacking all files is much more complex and takes longer (in terms of CPU and I/O needed) than many users expect. It's not a simple database update query.

**Comment by alexislefebvre** [comment]
Thanks for your answer. I use a readonly volume, without raw or xmp files, and only index files, so it looks like there would be no need to rename original files in that case.

**Comment by lastzero** [comment]
If files won't get stacked based on your file names at all, you should be good disabling stacking by metadata / id in Settings before you index.

**Comment by davepfrommer** [comment]
@lastzero 

I indexed/imported files with all the stacking options turned off; however, occasionally I will come across a still image which has an unrelated video attached/stacked.

Is there a way to list out which files (or in which folders) photos/videos are stacked?  
If there is a way to query the database or inspect the files somehow to determine where to look for stacked images, it would help quite a bit.

Thanks

**Comment by lastzero** [comment]
https://demo.photoprism.app/photos?q=stack%3Atrue

Videos with a preview image will always be stacks, just like live photos. So this page only shows stacked images & RAWs.

**Comment by ant0nwax** [comment]
stacking was turned off, and after an upgrade stacking is switched on again, how to batch unstack everything, stacking was never something i needed and i would like to avoid it to be present

**Comment by lastzero** [comment]
We are not aware of any reason settings should get lost when upgrading. Might be a problem with the storage folder / settings YAML file? Without knowing the exact steps to reproduce this, we can't help and investigate. Ideally use GitHub Discussions for this so not everyone gets an email for every new comment. Thank you!
